src/D2L.CodeStyle.Analyzers/Extensions/Microsoft.CodeAnalysis.cs

```diff
@@ -77,7 +77,14 @@ internal static partial class RoslynExtensions {
 			if( type.ContainingAssembly.TryGetImmutableGenericAnnotation( type, out ignore ) ) {
 				return true;
 			}
+
 			foreach( var typeArgument in type.TypeArguments ) {
+				// Can happen with generics, but I don't fully understand when
+				// it does, exactly.

```

Comment:
Left a comment because at some point I'll probably figure it out. the impact of `continue` here is that we don't consider something immutable which is safe at least.

Reply:
Ok, that's probably the better way to fail (too restrictive instead of too permissive)